index.js

```diff
@@ -18,6 +18,21 @@ var JSONReporter = function (baseReporterDecorator, config, helper, logger) {
     summary : {}
   };
 
+  let setFailedExamplesPerFile = function(failure) {
+    let title = failure.description
+    if (title.includes("hook for")) {
+      const title_regexp = new RegExp(".*hook for (.*)");
+      title = title_regexp.exec(title)[1];
+    }
+    const fullTitle = failure.suite.concat([title]).join(' ');
```

Comment:
This is the actual change

Reply:
Could you provide a diff for a test which produces the need for this change?

Reply:
Yes sure! If there is an error in a hook function, this is reported as a failure. In this case Mocha reports the hook test name which is `hook title + test title`, e.g.`"before all" hook for <test title>`.
I tried it by introducing a typo in a `before` hook.
The problem in such cases is that we use the reported test name to rerun only the failed examples (passing the names to Mocha `grep`), but in case of hook error the test name will not match the actual example name...

Reply:
Nice catch @gnanou, the fix LGTM!